src/probnum/filtsmooth/optim/_gauss_newton.py

```diff
@@ -0,0 +1,35 @@
+"""Gauss-Newton methods in state-space models."""
```

Comment:
I think it's good that you moved this out of `_kalman.py`, besides the other "regular" Kalman filter functionalities. 👍 

Do you think the name `optim` for the package could suggest that we optimize - for instance - hyperparameters for state-space model regression, or sth like that? Is it clear enough that the content of the package **is** itself an optimizer, rather than it **does** optimize sth.? (minor remark)

Reply:
I have a hunch that we might even want to think about moving hyperparameter optimisation in here. This stems from the notion that e.g. EM updates to, for instance, the initial distribution parameters, are a one-linear change to an IEKS. Perhaps we *want* to blur the lines here?! I dont know.